### PR TITLE
12.0 sale mrp set qty delivered

### LINKED ISSUE #32354: Set `qty_delivered` as `product_uom_qty` if not moves
https://github.com/odoo/odoo/blob/4df568549a685cb38cd66f6098e057ebee8be656/addons/sale_mrp/models/sale_mrp.py#L20

This line evaluate to `True` if not moves are in `line`

### PR DESCRIPTION
**Pull request by AfroMonkey**
[FIX] sale_order: Set qty_delivered to 0 if not moves.

In `_compute_qty_delivered` function
Python `all` function returns `True` if list is empty; thast why check 
if `moves` is not empty to set `qty_delivered` to `product_uom_qty`.

FIX #32354 

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by AfroMonkey** [comment]
FIX#32354

File: addons/sale_mrp/models/sale_mrp.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def _compute_qty_delivered(self):
                 if bom and bom.type == 'phantom':
                     moves = line.move_ids.filtered(lambda m: m.picking_id and m.picking_id.state != 'cancel')
                     bom_delivered = all([move.state == 'done' for move in moves])
```
Comment (fmdl) [inline addons/sale_mrp/models/sale_mrp.py]: I think it is better to do this :

`bom_delivered = moves and all([move.state == 'done' for move in moves])`

Comment (AfroMonkey) [reply addons/sale_mrp/models/sale_mrp.py]: Yes, it seems better

**Comment by amoyaux** [comment]
robodoo r+ rebase-ff

**Pull request closed without merging**